pkg/rbeconfigsgen/rbeconfigsgen.go

```diff
@@ -0,0 +1,932 @@
+		}
+	}
+	if len(o.OutputSourceRoot) != 0 {
+		if err := copyConfigsToOutputDir(o, oc); err != nil {
+			return fmt.Errorf("failed to write configs to directory %q: %w", o.OutputSourceRoot, err)
+		}
+	}
+	return nil
+}
+
+// digestFile returns the sha256 digest of the contents of the given file.
+func digestFile(filePath string) (string, error) {
+	f, err := os.Open(filePath)
+	if err != nil {
+		return "", fmt.Errorf("unable to open file %q: %w", filePath, err)
+	}
+	defer f.Close()
+
+	h := sha256.New()
+	if _, err := io.Copy(h, f); err != nil {
+		return "", fmt.Errorf("error while hashing the contents of %q: %w", filePath, err)
+	}
+	return hex.EncodeToString(h.Sum(nil)), nil
+}
+
+// createManifest writes a manifest text file containing information about the generated configs if
+// the given options specified a manifest file.
+func createManifest(o *Options) error {
+	if len(o.OutputManifest) == 0 {
+		return nil
+	}
+	f, err := os.Create(o.OutputManifest)
+	if err != nil {
+		return fmt.Errorf("unable to open a new file for writing manifest to %q: %w", o.OutputManifest, err)
+	}
+	defer f.Close()
+	fmt.Fprintf(f, "BazelVersion=%s\n", o.BazelVersion)
+	fmt.Fprintf(f, "ToolchainContainer=%s\n", o.ToolchainContainer)
+	// Extract the sha256 digest from the image name to be included in the manifest.
+	s := imageDigestRegexp.FindStringSubmatch(o.PlatformParams.ToolchainContainer)
+	if len(s) != 2 {
+		return fmt.Errorf("failed to extract sha256 digest using regex from image name %q, got %d substrings, want 2", o.PlatformParams.ToolchainContainer, len(s))
+	}
+	fmt.Fprintf(f, "ImageDigest=%s\n", s[1])
+	fmt.Fprintf(f, "ExecPlatformOS=%s\n", o.PlatformParams.OSFamily)
+	// Include the sha256 digest of the configs tarball if output tarball generation was enabled by
+	// actually hashing the contents of the output tarball.
+	if len(o.OutputTarball) != 0 {
+		d, err := digestFile(o.OutputTarball)
+		if err != nil {
+			return fmt.Errorf("unable to compute the sha256 digest of the output tarball file for the output manifest: %w", err)
+		}
+		fmt.Fprintf(f, "ConfigsTarballDigest=%s\n", d)
+	}
+	log.Printf("Wrote output manifest to %q.", o.OutputManifest)
+	return nil
+}
+
+// Run is the main entrypoint to generate Bazel toolchain configs according to the options
+// specified in the given command line arguments.
+// The file structure of the generated configs will be as follows:
+// <config root>
+// |
+//  - cc-  C++ configs as generated by Bazel's internal C++ toolchain detection logic.
+//  - config- Toolchain entrypoint target for cc_crosstool_top & the auto-generated platform target.
+//  - java- Java toolchain definition.
+func Run(o Options) error {
+	if err := processTempDir(&o); err != nil {
+		return fmt.Errorf("unable to initialize a local temporary working directory to store intermediate files: %w", err)
+	}
+	r, err := newDockerRunner(o.ToolchainContainer, o.Cleanup)
```

Comment:
Following up on the previous comment, could you switch this too from r to d?

Reply:
Done